### PR TITLE
feat: add github action for running tests

### PR DESCRIPTION
**Pull request by suhussai**
### Issue # (if applicable)

Closes #<issue number here>.

### Reason for this change

We need a mechanism for testing new changes.

### Description of changes

Adds a github action to conduct tests.

### Checklist

- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/awslabs/sbt-aws/blob/main/CONTRIBUTING.md)

---

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the project license.*

### CODE REVIEW

File: projenrc/run-tests-workflow.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    runTests.on({
+      pullRequest: {
+        branches: ['main'],
+      },
+    });
+
+    runTests.addJobs({
+      'run-tests': {
+        runsOn: ['ubuntu-22.04'],
+        if: "github.event.review.state == 'APPROVED'",
```
Comment (suhussai) [inline projenrc/run-tests-workflow.ts:19]: is this the right approach?

Comment (suhussai) [reply projenrc/run-tests-workflow.ts:19]: we will be going with this approach for now, but will review in the future.

**Review by tobuck-aws — APPROVED** [review]
A couple of comments here, one of which is repeated from above:

1. While I find these tests useful, they're not "unit-tests". If you look at all of the components that play a part in each of these tests, a good amount of them are actually NOT our components (and would've presumably been tested by AWS). That's not to say we don't need a test like this, because I do think it's valuable, but we should decide on which cadence. Is this something that should be run on every PR? Or daily? Basically, what cadence?
2. Do we need our CodeBuild wrapped in a StepFunction? I realize that if we connected directly CodeBuild, we run the risk of a PR providing its own CodeBuild file, which is obviously a potential security threat. That threat is mitigated using the StepFunction, but is the complexity worth the risk? We can discuss in our weekly meeting.

**Comment by suhussai** [comment]
Regarding item 1, I've updated the config to run the job nightly.

Regarding 2, we're planning on using the current approach for now.

File: scripts/github-actions-run-tests-script.sh
[the remark this answers is not in the document]
Comment (suhussai) [reply scripts/github-actions-run-tests-script.sh:1]: Responded in comment.

File: projenrc/run-tests-workflow.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+import { AwsCdkConstructLibrary } from 'projen/lib/awscdk';
+import { JobPermission } from 'projen/lib/github/workflows-model';
+
+export function runTestsWorkflow(project: AwsCdkConstructLibrary) {
+  const runTests = project.github?.addWorkflow('run-tests');
+  if (runTests) {
+    runTests.on({
+      schedule: [
+        { cron: '0 6 * * *' }, // Runs at midnight Mountain Time (UTC-6) every day
```
Comment (tobuck-aws) [inline projenrc/run-tests-workflow.ts:12]: Mountain time, eh?

**Review by tobuck-aws — APPROVED** [review]
Looks good to me!

File: projenrc/run-tests-workflow.ts (hunk 2 above)
Comment (suhussai) [reply projenrc/run-tests-workflow.ts:12]: Yeah, it was either east coast or west coast and mountain is kind of in the middle, so it's a little more fair 😁

**Pull request merged**